README.md

```diff
@@ -121,48 +119,44 @@ We'll also need to have the [Android SDK](https://developer.android.com/studio/#
-
-# android build configuration
-export PYTHON3_EXEC="$( which python3 )"
-export ANDROID_ABI=armeabi-v7a
-export ANDROID_NATIVE_API_LEVEL=android-21
-export ANDROID_TOOLCHAIN_NAME=arm-linux-androideabi-clang
-
-# pull and build ros2 for android
-mkdir -p ${ROS2_ANDROID_WORKSPACE}/src
-cd ${ROS2_ANDROID_WORKSPACE}
-curl https://raw.githubusercontent.com/ros2-java/ros2_java/dashing/ros2_java_android.repos
-vcs import ${ROS2_ANDROID_WORKSPACE}/src < ros2_java_android.repos
-source ${AMENT_WORKSPACE}/install_isolated/local_setup.sh
-ament build --isolated --skip-packages test_msgs \
-  --cmake-args \
-  -DPYTHON_EXECUTABLE=${PYTHON3_EXEC} \
-  -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
-  -DANDROID_FUNCTION_LEVEL_LINKING=OFF \
-  -DANDROID_NATIVE_API_LEVEL=${ANDROID_NATIVE_API_LEVEL} \
-  -DANDROID_TOOLCHAIN_NAME=${ANDROID_TOOLCHAIN_NAME} \
-  -DANDROID_STL=gnustl_shared \
-  -DANDROID_ABI=${ANDROID_ABI} \
-  -DANDROID_NDK=${ANDROID_NDK} \
-  -DTHIRDPARTY=ON \
-  -DCOMPILE_EXAMPLES=OFF \
-  -DCMAKE_FIND_ROOT_PATH="$AMENT_WORKSPACE/install_isolated;$ROS2_ANDROID_WORKSPACE/install_isolated" \
-  -- \
-  --parallel \
-  --ament-gradle-args \
-  -Pament.android_stl=gnustl_shared -Pament.android_abi=$ANDROID_ABI -Pament.android_ndk=$ANDROID_NDK --
-```
+1. Download the [Android NDK](https://developer.android.com/ndk/downloads/index.html) and set the environment variable `ANDROID_NDK` to the path where it is extracted.
+
+1. Download the [Android SDK](https://developer.android.com/studio/#downloads) and set the environment variable `ANDROID_HOME` to the path where it is extracted.
+
+1. Clone ROS 2 and ROS 2 Java source code:
+
+        mkdir -p $HOME/ros2_android_ws/src
+        cd $HOME/ros2_android_ws
+        curl https://raw.githubusercontent.com/ros2-java/ros2_java/dashing/ros2_java_android.repos | vcs import src
+
+1. Set Android build configuration:
+
+        export PYTHON3_EXEC="$( which python3 )"
+        export ANDROID_ABI=armeabi-v7a
+        export ANDROID_NATIVE_API_LEVEL=android-21
+        export ANDROID_TOOLCHAIN_NAME=arm-linux-androideabi-clang
+
+1. Build (skipping packages that we don't need or can't cross-compile):
+
+        colcon build --packages-skip test_msgs \
+          --cmake-args \
+          -DPYTHON_EXECUTABLE=${PYTHON3_EXEC} \
+          -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
+          -DANDROID_FUNCTION_LEVEL_LINKING=OFF \
+          -DANDROID_NATIVE_API_LEVEL=${ANDROID_NATIVE_API_LEVEL} \
+          -DANDROID_TOOLCHAIN_NAME=${ANDROID_TOOLCHAIN_NAME} \
+          -DANDROID_STL=c++_shared \
+          -DANDROID_ABI=${ANDROID_ABI} \
+          -DANDROID_NDK=${ANDROID_NDK} \
+          -DTHIRDPARTY=ON \
+          -DCOMPILE_EXAMPLES=OFF
+
+          # TODO: These options may still need to be adapted
+          # -DCMAKE_FIND_ROOT_PATH="$AMENT_WORKSPACE/install_isolated;$ROS2_ANDROID_WORKSPACE/install_isolated" \
```

Comment:
I think this is what is causing CMake not to be able to find the `ament` packages, I'll compile this PR with this enabled and hopefully it should work.